bin/new-git-repository

```diff
@@ -1,97 +1,121 @@
-        --private)
-            PUBPRIV=private
-            shift
-            ;;
-        -g|--group)
-            shift
-            if test $# -gt 0; then
-                GROUP=$1
-            else
-                echo "No group name specified"
-                exit 1
-            fi
-            shift
-            ;;
-        *)
-            NAME=$1
-            shift
-            ;;
-    esac
-done
-
-if [ x$NAME = x -o x$PUBPRIV = x ]; then
-  echo "You must give a repository name, and --public or --private\n"
-  help
-  exit 1
-fi
-
-if [ $PUBPRIV = private -a x$GROUP = x ]; then
-  GROUP=privatecvs
-elif [ $PUBPRIV = public ]; then
-  GROUP=publiccvs
-fi
-
-NAME=${NAME%.git}
-NAME=$NAME.git
-
-if [ -e /data/git/$PUBPRIV/$NAME ]; then
-  echo "That repository name already exists"
-  exit 1
-fi
-
-cd /data/git/$PUBPRIV
-sudo mkdir $NAME
-cd $NAME
-sudo git init --quiet --bare --shared=group
-cd ..
-sudo chgrp -R $GROUP $NAME
-if [ $PUBPRIV = private ]; then
-  sudo chmod -R o-xr $NAME
-fi
-cd $NAME/hooks/
-ln -s /data/mysociety/bin/git-post-receive-hook post-receive
-cd ..
-git config hooks.emailprefix ""
-git config hooks.mailinglist git-$PUBPRIV@mysociety.org
-git config hooks.showrev "git show -C %s; echo"
-git config receive.denynonfastforwards false
-if [ $PUBPRIV = public ]; then
-  touch git-daemon-export-ok
-fi
-
-echo "New repository created at /data/git/$PUBPRIV/$NAME"
+#!/usr/bin/python
+
+import argparse
+import json
+import os
+import requests
+import socket
+import subprocess
+import sys
+
+def is_correct_server():
+    git_ip = socket.gethostbyname('git.mysociety.org')
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.connect(('46.43.39.78', 0))  # connecting to a UDP address doesn't send packets
+    local_ip = s.getsockname()[0]
+
+    if git_ip != local_ip:
+        sys.exit("You can only run this on leopard")
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Create a new git repository under /data/git')
+    parser.add_argument('name')
+    group1 = parser.add_argument_group()
+    group2 = group1.add_mutually_exclusive_group(required=True)
+    group2.add_argument('--public', action='store_const', const='public', dest='pubpriv', help='Public repository')
+    group2.add_argument('--private', action='store_const', const='private', dest='pubpriv', help='Private repository')
+    group1.add_argument('--group', help='Group name if private (defaults to privatecvs)')
+    parser.add_argument('--github', action='store_true', help='Create repo on GitHub')
+    args = parser.parse_args()
+
+    if args.pubpriv == 'public':
+        if args.group:
+            parser.error('Do not specify a group for a public repository')
+        args.group = 'publiccvs'
+    else:
+        if not args.group:
+            args.group = 'privatecvs'
+
+    if args.name[-4:] != '.git':
+        args.name += '.git'
+
+    return args
+
+def sudo(*args):
+    subprocess.call(['sudo'] + list(args))
```

Comment:
This may just be me, but I don't like it when scripts run `sudo` on my behalf, since it might do something as root silently if I'd recently entered my password when running another command as `sudo`. My preference is for a script that needs elevated privileges to check if it's running as root at the start and error if not, so you can re-run with `sudo` if you're happy that with that.